app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontOpenProductEntityPageActionGroup.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontOpenProductEntityPageActionGroup">
+        <annotations>
+            <description>Opens Storefront Product page for the provided Product Entity</description>
+        </annotations>
+        <arguments>
+            <argument name="product" type="entity"/>
+        </arguments>
+
+        <amOnPage url="{{StorefrontProductPage.url(product.custom_attributes[url_key])}}" stepKey="goToProductPage"/>
```

Comment:
This one was added to load the Product URL properly using `StorefrontProductPage.url` and `url_key` instead of Product's `name` :man_facepalming: